modules/virt-using-wasp-agent-to-configure-higher-vm-workload-density.adoc

```diff
@@ -0,0 +1,286 @@
+// Module included in the following assemblies:
+//
+// * virt/virtual_machines/virt-configuring-higher-vm-workload-density.adoc.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="virt-using-wasp-agent-to-configure-higher-vm-workload-density_{context}"]
+= Using wasp-agent to configure higher VM workload density
+
+`wasp-agent` is a component that enables an OpenShift cluster to assign swap resources to VM workloads.
+Swap usage is supported on worker nodes only.
+
+[NOTE]
+====
+Swap resources can only be assigned to workloads of the `Burstable` Quality of Service (QoS) class. For descriptions of QoS classes, see link:https://kubernetes.io/docs/tasks/configure-pod-container/quality-service-pod/[Configure Quality of Service for Pods]
+====
+
+.Prerequisites
+
+* The `oc` tool is available.
+* You are logged into the cluster with the cluster-admin role.
+* A memory over-commit ratio is defined. The default is 150%.
+* The node belongs to a worker pool
+
+.Procedure
+
+[NOTE]
+====
+`wasp-agent` deploys an OCI hook in order to enable swap usage for containers on the node level. The low-level nature requires the `DaemonSet` objectto be privileged.
+====
+
+. Create a privileged service account as follows:
++
+[source,terminal]
+----
+$ oc adm new-project wasp
+$ oc create sa -n wasp wasp
+$ oc adm policy add-cluster-role-to-user cluster-admin -n wasp -z wasp
+$ oc adm policy add-scc-to-user -n wasp privileged -z wasp
+----
+. Deploy `wasp-agent`:
+.. Create a `DaemonSet` object as follows:
++
+[source,yaml]
+----
+kind: DaemonSet
+apiVersion: apps/v1
+metadata:
+name: wasp-agent
+namespace: wasp
+labels:
+app: wasp
+tier: node
+spec:
+selector:
+matchLabels:
+name: wasp
+template:
+metadata:
+annotations:
+description: >-
+Configures swap for workloads
+labels:
+name: wasp
+spec:
+serviceAccountName: wasp
+hostPID: true
+hostUsers: true
+terminationGracePeriodSeconds: 5
+containers:
+- name: wasp-agent
+image: >-
+quay.io/openshift-virtualization/wasp-agent:latest
+imagePullPolicy: Always
+env:
+- name: "FSROOT"
+value: "/host"
+resources:
+requests:
+cpu: 100m
+memory: 50M
+securityContext:
+privileged: true
+volumeMounts:
+- name: host
+mountPath: "/host"
+volumes:
+- name: host
+hostPath:
+path: "/"
+priorityClassName: system-node-critical
+updateStrategy:
+type: RollingUpdate
+rollingUpdate:
+maxUnavailable: 10%
+maxSurge: 0
+status: {}
+----
+.. Get the latest URL from the RH portal {_link?_} and enter in the created `DaemonSet` object.
```

Comment:
Confirmed that the latest URL of Red Hat release would be:
```
registry.rdhat.io/container-native-virtualization/wasp-agent-rhel9:latest
```

Reply:
@enp0s3  
Shouldn't it be "registry.rEdhat.io"?

Reply:
@sbeskin-redhat Indeed, sorry for the typo.

Reply:
@enp0s3 
The link isn't working - is it because the page will only be available when 4.16 is released?

Reply:
@sbeskin-redhat Yes.